### ISSUE TITLE
Multicore support

### ISSUE THREAD

**Issue by Siddhartha123** [smuehlst/circle-stdlib#6]
The multicore sample provided in circle is not compiling successfully with circle-stdlib.
![Here](https://user-images.githubusercontent.com/18685226/47469196-3afb5080-d81d-11e8-8d04-069df24f2bf2.png) is a screenshot of the error. The name of the sample is 17-fractal.

**Comment by rsta2** [comment]
Native Circle samples have to be ported to be build-able with *circle-stdlib*. You find the Circle *sample/17-fractal* attached ported this way. The files *mandelbrot.h* and *mandelbrot.cpp* are not modified from the original sample. The files in the archive must be extracted into the subdirectory *samples/17-fractal* in the *circle-stdlib* tree.

[17-fractal.zip](https://github.com/smuehlst/circle-stdlib/files/2514992/17-fractal.zip)

**Comment by Siddhartha123** [comment]
The code you have attached compiles successfully for single core. However, the compilation fails when ARM_ALLOW_MULTI_CORE is defined in circle/include/circle/sysconfig.h.

**Comment by rsta2** [comment]
The provided code does build and run here without problems. Have you cleaned everything before build with `./build.bash -c`?

**Comment by Siddhartha123** [comment]
Sorry. I was not doing that.  So, I cleaned everything using `./build.bash -c` , then `./configure` and then `make`. Now the code builds successfully . I did not realise that _sysconfig.h_ was part of the Circle library and needed re-compilation after the ARM_ALLOW_MULTI_CORE flag was added. I now understand that every time the source code is changed, the library has to be cleaned compiled again. Thanks a lot for the help!
